### PR TITLE
Added UntilNext method which provides a channel to block

### PR DESCRIPTION
**Pull request by voldyman**

### CODE REVIEW

File: limiter.go [hunk 1]
```diff
@@ -7,14 +7,16 @@ import (
...
+// ErrRateExceeded is the error returned when the read rate exceeds our specification.
 var ErrRateExceeded = errors.New("Read rate exceeded.")
 
 // Limiter is an interface for a rate limiter.
 // There are a few example limiters included in the package, but feel free to go wild with your own.
 type Limiter interface {
 	// Apply this many bytes to the limiter, return ErrRateExceeded if the defined rate is exceeded.
 	Count(int) error
+	// UntilNext returns a channel that can be used to block until next window
+	UntilNext() <-chan time.Time
```
Comment (shazow) [inline limiter.go]: Might be nicer to not add this to the full Limiter interface but just on the concrete type, and change the constructors to return the concrete type reference instead (should have done that originally--accept interfaces, return concrete types)

File: limiter.go [hunk 2]
```diff
@@ -60,3 +62,12 @@ func (limit *simpleLimiter) Count(n int) error {
 	}
 	return nil
 }
+
+// UntilNext returns a channel that can be used to block until next window
+func (limit *simpleLimiter) UntilNext() <-chan time.Time {
```
Comment (shazow) [inline limiter.go]: Delay() is maybe a better name but I don't hate this one either. :)

File: limiter_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	limiter := NewSimpleLimiter(10, 10*time.Second)
+	if err := limiter.Count(1); err != nil {
+		t.Fatal("got rate limited on adding the first 'Count'")
+	}
+
+	before := time.Now()
+	waitChan := limiter.UntilNext()
+	after := <-waitChan
+
+	if after.Sub(before) > 500*time.Millisecond {
```
Comment (shazow) [inline limiter_test.go:44]: This is probably the saddest thing of it all. :P Oh well. Sad tests are better than no tests.

**Review by shazow — APPROVED** [review]
Ready to merge when you say so. :)

Also another nice-to-have for a future PR: Make Count() threadsafe by using sync/atomic.

File: limiter.go (hunk 1 above)
Comment (voldyman) [reply limiter.go]: done

File: limiter.go (hunk 2 above)
Comment (voldyman) [reply limiter.go]: changed to delay. :)

**Pull request merged**